api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -1,236 +1,141 @@
-        return conversionCoefficient;
-    }
+    void to(@NotNull Currency currency, double amount, @NotNull EconomySubscriber<Double> subscription);
 
     /**
-     * Sets a new conversion coefficient of this currency.
+     * Used to get the double representation of an amount represented by a formatted string.
      *
-     * @param conversionCoefficient new conversion coefficient
-     * @author MrIvanPlays
+     * @param formatted The formatted string to be converted to double form.
+     * @param subscription The {@link EconomySubscriber} accepting the resulting {@link Double} that
+     *                     represents the deformatted amount of the formatted String.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    protected void setConversionCoefficient(double conversionCoefficient) {
-        this.conversionCoefficient = conversionCoefficient;
-    }
+    void parse(@NotNull String formatted, @NotNull EconomySubscriber<Double> subscription);
 
     /**
-     * Get the starting balance of the currency.
-     * The player {@link UUID} is nullable, it should be specified if the starting balance
-     * concerns a {@link me.lokka30.treasury.api.economy.account.PlayerAccount}.
+     * Gets the starting balance of a specific player account for this currency.
      *
-     * @param playerUUID a UUID of the player account created. For global scenarios, specify `null`.
-     * @return the starting balance of the currency concerning specified player's UUID.
-     * @author lokka30, Geolykt, MrIvanPlays
+     * @param playerID The UUID of the player we are getting the starting balance for.
+     * @return The starting balance of the player for this currency.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    public double getStartingBalance(@Nullable UUID playerUUID) {
-        return startingBalance.apply(playerUUID);
-    }
+    double getStartingBalance(@Nullable UUID playerID);
 
     /**
-     * Gets a human-readable format of the balance.
-     * For example, '$1.50' or '1.50 dollars' or 'One dollar and fifty cents'.
+     * Used to translate an amount to a user readable format with the default precision.
      *
-     * @param amount to be formatted.
-     * @param locale of the formatted balance being requested.
-     * @return the human-readable format of the specified amount and locale.
-     * @author lokka30, MrIvanPlays
+     * @param amount The amount to format.
+     * @return The formatted text.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @NotNull
-    public String formatBalance(double amount, @NotNull Locale locale) {
-        Objects.requireNonNull(locale, "locale");
-        return balanceFormatter.apply(amount, locale);
-    }
+    String format(double amount);
```

Comment:
I think this should also accept a `@Nullable Player player` as depending on the player who is going to be shown the amount diferent strings will be generated. For example: `1.000.254,125` or `1,000,254.125` or the provider allows each player to decide the format that the amount is shown and the string ends up being `1 000 254.125` for player A and `1M` for player B

Reply:
Ooh, good spotting. Looks like the new method doesn't have the Locale argument. Should we use the Locale or Player? Player being handled by the provider of course.